constants/item_constants.asm

```diff
@@ -102,58 +102,82 @@ SAFARI_ROCK           EQU $16 ; overload
 
 const_value = $C4
 
-	const HM_01         ; $C4
-	const HM_02         ; $C5
-	const HM_03         ; $C6
-	const HM_04         ; $C7
-	const HM_05         ; $C8
-	const TM_01         ; $C9
-	const TM_02         ; $CA
-	const TM_03         ; $CB
-	const TM_04         ; $CC
-	const TM_05         ; $CD
-	const TM_06         ; $CE
-	const TM_07         ; $CF
-	const TM_08         ; $D0
-	const TM_09         ; $D1
-	const TM_10         ; $D2
-	const TM_11         ; $D3
-	const TM_12         ; $D4
-	const TM_13         ; $D5
-	const TM_14         ; $D6
-	const TM_15         ; $D7
-	const TM_16         ; $D8
-	const TM_17         ; $D9
-	const TM_18         ; $DA
-	const TM_19         ; $DB
-	const TM_20         ; $DC
-	const TM_21         ; $DD
-	const TM_22         ; $DE
-	const TM_23         ; $DF
-	const TM_24         ; $E0
-	const TM_25         ; $E1
-	const TM_26         ; $E2
-	const TM_27         ; $E3
-	const TM_28         ; $E4
-	const TM_29         ; $E5
-	const TM_30         ; $E6
-	const TM_31         ; $E7
-	const TM_32         ; $E8
-	const TM_33         ; $E9
-	const TM_34         ; $EA
-	const TM_35         ; $EB
-	const TM_36         ; $EC
-	const TM_37         ; $ED
-	const TM_38         ; $EE
-	const TM_39         ; $EF
-	const TM_40         ; $F0
-	const TM_41         ; $F1
-	const TM_42         ; $F2
-	const TM_43         ; $F3
-	const TM_44         ; $F4
-	const TM_45         ; $F5
-	const TM_46         ; $F6
-	const TM_47         ; $F7
-	const TM_48         ; $F8
-	const TM_49         ; $F9
-	const TM_50         ; $FA
+add_hm: MACRO
+if !DEF(HM01)
+HM01 EQU const_value
+	enum_start 51 ; NUM_TMS + 1
+endc
+	const HM_\1
+	enum \1_TMNUM
+ENDM
+
+	add_hm CUT          ; $C4
+	add_hm FLY          ; $C5
+	add_hm SURF         ; $C6
+	add_hm STRENGTH     ; $C7
+	add_hm FLASH        ; $C8
+NUM_HMS EQU const_value - HM01
+
+add_tm: MACRO
+if !DEF(TM01)
+TM01 EQU const_value
```

Comment:
Am I understanding this line correctly (and line 107) that `TM01` (and `HM01`) is declared, but none of the others? (Not critiquing, just making sure I'm reading it right.)

Reply:
Yes, they're meant for marking the first TM and first HM.